### PR TITLE
UDP sockect doc update

### PR DESCRIPTION
**Pull request by deepikabhavnani**
Jira: https://jira.arm.com/browse/IOTMORF-1309

Added the example reference of UDP client

@sg- Please review

### CODE REVIEW

**Comment by deepikabhavnani** [comment]
@sg- Please review

File: docs/reference/api/connectivity/networksocket/UDPSocket.md [hunk 1]
```diff
@@ -2,6 +2,14 @@
 
 The UDPSocket class provides the ability to send packets of data over UDP, using the `sendto` and `recvfrom` member functions. Packets can be lost or arrive out of order, so we suggest using a [TCPSocket](/docs/v5.6/reference/tcpsocket.html) when you require guaranteed delivery.
 
+The constructor takes in the [NetworkStack](/docs/v5.6/mbed-os-api-doxy/class_network_stack.html) or [NetworkInterface](/docs/v5.6/mbed-os-api-doxy/class_network_interface.html) pointer to open the socket on specified NetworkInterface. If NetworkInterface is not passed in constructor, than you must call `open` to initialize the socket.
```
Comment (sg-) [inline docs/reference/api/connectivity/networksocket/UDPSocket.md]: same comments as TCP

File: docs/reference/api/connectivity/networksocket/UDPSocket.md [hunk 2]
```diff
@@ -2,6 +2,14 @@
 
 The UDPSocket class provides the ability to send packets of data over UDP, using the `sendto` and `recvfrom` member functions. Packets can be lost or arrive out of order, so we suggest using a [TCPSocket](/docs/v5.6/reference/tcpsocket.html) when you require guaranteed delivery.
 
+The constructor takes in the NetworkStack pointer to open the socket on the specified NetworkInterface. If NetworkStack is not passed in the constructor, then you must call `open` to initialize the socket.
```
Comment (AnotherButler) [inline docs/reference/api/connectivity/networksocket/UDPSocket.md]: Query: Who or what passes in the constructor?

Comment (deepikabhavnani) [reply docs/reference/api/connectivity/networksocket/UDPSocket.md]: NetworkStack pointer

**Pull request merged**